### PR TITLE
Allow documentation tests in FITS verification page

### LINKED ISSUE #6686: FITS verification documentation is still untested
The page on Verification of `io.fits` (http://docs.astropy.org/en/latest/io/fits/usage/verification.html) wasn't changed in #6664 and many parts don't work at all.

Related: #6010

### PR DESCRIPTION
**Pull request by cmarmo**
### Description
This pull request removes the `.. doctest-skip` directive from the [FITS verification page](http://docs.astropy.org/en/latest/io/fits/usage/verification.html). I could not find a better way to manage warnings than inserting a [doctest.IGNORE_EXCEPTION_DETAIL](https://docs.python.org/3/library/doctest.html#doctest.IGNORE_EXCEPTION_DETAIL).

Fixes #6686.

### CODE REVIEW

File: docs/io/fits/usage/verification.rst
Comment (saimn) [inline docs/io/fits/usage/verification.rst:263]: Hmm, it's a bit unfortunate that we turn warnings into errors (for unit tests) here, since for users this should give only a warning by default. I see two solutions:
- we could maybe use the `testsetup` and `testcleanup` directives to remove the 'error' filter before running this file,
- or just wrap this line with `warning.catch_warnings` to make sure warnings are not turn into exceptions here.

File: docs/io/fits/usage/verification.rst
Comment (saimn) [inline docs/io/fits/usage/verification.rst:288]: Same issue with warnings.

File: docs/io/fits/usage/verification.rst
Comment (saimn) [inline docs/io/fits/usage/verification.rst:354]:
```suggestion
    >>> filename = ...
```
And same below.

**Review by saimn — COMMENTED** [review]
Thanks for your work on this @cmarmo, very useful since obviously some examples were out of date. I'm not how to best handle the warnings/exception issue (some ideas in the comments). It would be better if the users can see the warnings as they would get them by default.

**Comment by pllim** [comment]
Re: Warnings -- If you want them to show but don't want doctest to fail, use https://github.com/astropy/pytest-doctestplus#ignoring-warnings

`# doctest: +IGNORE_WARNINGS`

**Comment by saimn** [comment]
@pllim - `IGNORE_WARNINGS` is to hide the warning isn't it ? I would see to see it, since that's the result of the FITS verification.

**Comment by pllim** [comment]
Yes, correct that `IGNORE_WARNINGS` does hide it... 🤔

File: docs/io/fits/usage/verification.rst
Comment (cmarmo) [reply docs/io/fits/usage/verification.rst]: Indeed... :)

**Comment by cmarmo** [comment]
Thanks @saimn and @pllim for your suggestions. I have recorded the warnings and printed them: note that one of the warnings is thrown as an exception by default, I've kept the `IGNORE_EXCEPTION_DETAIL` directive for that one. I'm not sure this is the most readable solution...

**Comment by cmarmo** [comment]
I've tried to clean a bit, but now to avoid `ResourceWarning: unclosed file _io.FileIO` warnings all fits files should be closed.

**Comment by saimn** [comment]
@cmarmo : I've hijacked a bit your PR, sorry, trying to find a better solution for warnings. To keep the docs clear and focused on the initial topic, it would be better if we could avoid all the warnings related code. So currently I have factored the catching and printing in a `showwarnings` context manager, which works well. And then I think we could move this to [pytest-doctestplus](https://github.com/astropy/pytest-doctestplus) in a new doctest flag.

**Comment by cmarmo** [comment]
> @cmarmo : I've hijacked a bit your PR, sorry, trying to find a better solution for warnings. To keep the docs clear and focused on the initial topic, it would be better if we could avoid all the warnings related code. So currently I have factored the catching and printing in a `showwarnings` context manager, which works well.

@saimn no problem, this kind of refactoring was above my python skills... always happy to learn... :)

> And then I think we could move this to [pytest-doctestplus](https://github.com/astropy/pytest-doctestplus) in a new doctest flag.

Do you mind if I add a comment in the testsetup block, to document that the function is accessory to the warning management in the example? and that it could one day be replaced by a `SHOW_WARNING` options?

**Comment by saimn** [comment]
> Do you mind if I add a comment in the testsetup block, to document that the function is accessory to the warning management in the example? and that it could one day be replaced by a SHOW_WARNING options?

@cmarmo - I managed to make a PR on doctestplus (https://github.com/astropy/pytest-doctestplus/pull/136), so if you don't mind waiting a bit we can update the PR once it's merged and released ? (hopefully soon!)

**Comment by cmarmo** [comment]
> update the PR once it's merged and released ?

Arrgggh, I should have wait for the release too... :(

**Comment by saimn** [comment]
I released the new version of pytest-doctestplus, and closed/reopened the PR to trigger the actions builds.

**Review by saimn — APPROVED** [review]
Seems all good now with the new doctestplus version. Thanks @cmarmo !

**Comment by cmarmo** [comment]
Thank you @saimn .

**Pull request merged**
